app/models/comment.rb

```diff
@@ -1,2 +1,11 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, foreign_key: 'post_id'
+
+  private
+
+  def update_comments_counter
+    post.increment!(:comments_counter)
+  end
```

Comment:
- **[OPTIONAL]** We can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new comment has been saved to database, then make the update_comments_counter method private 👇🏽 

```ruby
after_save :update_comments_counter

  private

  def update_comments_counter
    post.increment!(:comments_counter)
  end
```